src/main/java/org/junit/experimental/results/ResultMatchers.java

```diff
@@ -34,14 +36,21 @@ public ResultMatchers() {
      * Matches if there are {@code count} failures
      */
     public static Matcher<PrintableResult> failureCountIs(final int count) {
+        return failureCount(equalTo(count));
+    }
+
+    /**
+     * Matches if the number of failures matches {@code countMatcher}
+     */
+    public static Matcher<PrintableResult> failureCount(final Matcher<Integer> countMatcher) {
```

Comment:
I'm curious why we need an overload that takes a `Matcher<Integer>`. Can you give an example of where this would be more useful than the version that takes an `int`?

Also, shouldn't the parameter have a type of `Matcher<? super Integer>` (see http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matchers.html#arrayWithSize(org.hamcrest.Matcher) for one example in Hamcrest)